File: Frontend/src/app/marks/interfaces/mark.ts
import { Block } from "./block";
import { Link } from "./link";

export interface Mark {
    id: number;
    name: string;
    userId: string;
    collectionId: number;
    createdDate?: Date;
    links?: Link[];
    blocks: Block[];
    emoji?: string;
    inputName?: string;
    collectionName?: string;
    requiresSync?: boolean;
}

File: Backend/markitAPI/markit.Domain/Entities/Mark.cs
using markit.Domain.Common;
using System.ComponentModel.DataAnnotations;

namespace markit.Domain.Entities
{
    public class Mark : BaseModel
    {
        [Required, MaxLength(255)] public string Name { get; set; } = string.Empty;
        [Required] public int CollectionId { get; set; }
        public bool IsFavorite { get; set; }
        public string? DocumentId { get; set; }
        public DateTime? LastSync { get; set; }
        public string? Emoji { get; set; }
        public bool? NameLess { get; set; }

        public virtual Collection? Collection { get; set; }
        public virtual ICollection<Link>? Links { get; set; }
        public virtual ICollection<Block>? Blocks { get; set; }
    }
}